src/core_plugins/metrics/public/services/__tests__/executor_provider.js

```diff
@@ -59,13 +59,17 @@ describe('$executor service', () => {
 
   it('should execute function if ingorePause is passed (interval set to 1000ms)', (done) => {
     timefilter.refreshInterval.value = 1000;
-    executor.register({ execute: () => done() });
```

Comment:
__note:__ `execute` should return `Promise`, locally it fails from time to time with `then method is undefined` or something like this.